new-log-viewer/src/utils/config.ts

```diff
@@ -0,0 +1,128 @@
+import {PAGE_SIZE} from "../contexts/StateContextProvider";
+import {
+    CONFIG_NAME,
+    ConfigMap,
+    ConfigUpdate,
+    LOCAL_STORAGE_KEY,
+} from "../typings/config";
+
+
+const DECODER_OPTIONS_DEFAULT = {
+    formatString: "%d{yyyy-MM-dd HH:mm:ss.SSS} [%process.thread.name] %log.level" + " %message%n",
+    logLevelKey: "log.level",
+    timestampKey: "@timestamp",
+};
+const THEMES = ["light",
+    "dark"];
+const THEME_DEFAULT = "light";
+
+/**
+ * Retrieves the configuration value from local storage based on the provided configuration name.
+ *
+ * @template T - The type of the configuration name.
+ * @param code The configuration name to retrieve the value for.
+ * @return - The configuration value or null if not found.
+ */
+const getConfig = <T extends CONFIG_NAME>(code: T): ConfigMap[T] | null => {
+    switch (code) {
+        case CONFIG_NAME.DECODER_OPTIONS:
+            return {
+                formatString: window.localStorage.getItem(
+                    LOCAL_STORAGE_KEY.DECODER_OPTIONS_FORMAT_STRING
+                ),
+                logLevelKey: window.localStorage.getItem(
+                    LOCAL_STORAGE_KEY.DECODER_OPTIONS_LOG_LEVEL_KEY
+                ),
+                timestampKey: window.localStorage.getItem(
+                    LOCAL_STORAGE_KEY.DECODER_OPTIONS_TIMESTAMP_KEY
+                ),
+            } as ConfigMap[T];
+        case CONFIG_NAME.THEME:
+            return window.localStorage.getItem(LOCAL_STORAGE_KEY.THEME) as ConfigMap[T];
+        case CONFIG_NAME.PAGE_SIZE:
+            return window.localStorage.getItem(LOCAL_STORAGE_KEY.PAGE_SIZE) as ConfigMap[T];
```

Comment:
Due to TypeScript's inability to validation the return type within the function body, the proposed approach is essentially a hack and a bit unclean. This is not implemented for now to avoid confusions.